python/src/iceberg/schema.py

```diff
@@ -262,10 +267,52 @@ def primitive(self, primitive: PrimitiveType) -> T:
         ...  # pragma: no cover
 
 
+@dataclass(init=True, eq=True)
+class Accessor:
+    """An accessor for a specific position in a container that implements the StructProtocol"""
+
+    position: int
+    inner: Optional["Accessor"] = None
+
+    def __str__(self):
+        return f"Accessor(position={self.position},inner={self.inner})"
+
+    def __repr__(self):
+        return self.__str__()
+
+    def with_inner(self, accessor: "Accessor") -> "Accessor":
```

Comment:
Is there a better name for this method? It sounds like it should replace this accessor's `inner` field, but it actually sets the leaf `inner`. Maybe `with_inner_most`?

Reply:
We don't need this one anymore :)